doc: add a little warning for <param ... multiple="true"> (#930)
galaxyproject/planemo: docs/_writing_collections.rst

-------------------------------

The ``data_collection`` parameter type can specify a ``collection_type`` or
``list`` but whenever possible, it might actually be better to not explicitly
consume lists as a tool author. Parameters of type ``data`` can include a
``multiple="True"`` attribute to allow many datasets to be selected
simultaneously. While the default UI will then have Galaxy users pick
individual datasets, they can choose a collections as the tool can
process both. This has the benefit of allowing tools to
process either individual datasets or collections.
A noteworthy difference is that if ``multiple="True"`` is used the elements of

> the collection are passed to the tool as (python) list, i.e., it is not

Reviewer: ```suggestion
the collection are passed to the tool as a (Python) list, i.e. it is not
```

-------------------------------

The ``data_collection`` parameter type can specify a ``collection_type`` or
``list`` but whenever possible, it might actually be better to not explicitly
consume lists as a tool author. Parameters of type ``data`` can include a
``multiple="True"`` attribute to allow many datasets to be selected
simultaneously. While the default UI will then have Galaxy users pick
individual datasets, they can choose a collections as the tool can
process both. This has the benefit of allowing tools to
process either individual datasets or collections.
A noteworthy difference is that if ``multiple="True"`` is used the elements of
the collection are passed to the tool as a (Python) list, i.e. it is not

> possible

Reviewer: ```suggestion
possible:
```

- to find out if a collection was passed, 
- to access properties of the collection (name,...), or

> - to write tests that pass collections to the parameter (which would allow to name the elements explicitly) 

Reviewer: ```suggestion
- to write tests that pass a collection to the parameter (which would allow to name the elements explicitly).
```

-------------------------------

The ``data_collection`` parameter type can specify a ``collection_type`` or

> ``list`` but whenever possible, it might actually be better to not explicitly

Reviewer 2: This represented an IUC recommendation... I think I agree with the language changes here - but I was always more pro-collection inputs then everyone else. So I'm nervous about making this less of a recommendation.

Author: Agreed. I changed the wording to "it is recommended".

consume lists as a tool author. Parameters of type ``data`` can include a
``multiple="True"`` attribute to allow many datasets to be selected
simultaneously. While the default UI will then have Galaxy users pick
individual datasets, they can choose a collections as the tool can
process both. This has the benefit of allowing tools to 
process either individual datasets or collections.

> A noteworthy difference is that if a parameter of type ``data`` wth ``multiple="True"`` is used the elements of

Reviewer: ```suggestion
A noteworthy difference is that if a parameter of type ``data`` with ``multiple="true"`` is used, the elements of
```

the collection are passed to the tool as a (Python) list, i.e. it is not
possible:

- to find out if a collection was passed, 
- to access properties of the collection (name,...), or
- to write tests that pass a collection to the parameter (which would allow to name the elements explicitly).

> Another drawback is that the `${on_string}` of the label contains the list of data sets in the collection (which can be confusing, since these data sets are in most cases hidden) and not the name of the collection. 

Reviewer: That's probably something which we may want to fix?
